doc/en/interactive.rst

```diff
@@ -33,37 +33,51 @@ interrupt processes started as part of the test environment.
-Because of this, it is recommended that test suites are defined in seperate modules.
-Another known limitation is that we are not able to reload module that is under
-namespace package, and the workaround is to add dummy __init__.py file.
+Code-reloading is triggered by reload API request, and there is a reload button
+on interactive mode UI. By default, testplan will look under the directory of
+the main script for modules to reload - those who have either been modified or
+have had its dependencies reloaded. User can instruct testplan to look further
+by using ``extra_deps`` parameter of ``@test_plan`` decorator.
 
 The steps of the reload process are described below:
 
-* When starting up the interactive mode, testplan builds a graph of dependencies used by the ``__main__`` module. Only dependencies that are within the same directory as the testplan script are included.
-  Dependencies are resolved using `modulefinder.ModuleFinder <https://docs.python.org/3/library/modulefinder.html#modulefinder.ModuleFinder>`_
+* When starting up the interactive mode, testplan builds a graph of dependencies
+  used by the ``__main__`` module. Only dependencies that are within the same
+  directory as the testplan script are included. Dependencies are resolved using
+  `modulefinder.ModuleFinder <https://docs.python.org/3/library/modulefinder.html#modulefinder.ModuleFinder>`_
 * Dependencies are checked and reloaded on reload API. When reloading modules testplan will:
 
   * Walk the graph of dependencies starting from ``__main__``.
-  * For each module, if any of its dependencies were reloaded or the file has been modified since last reloaded, then reload this module and update __class__ of all suites it defines.
+  * For each module, if any of its dependencies were reloaded or the file has
+    been modified since last reloaded, then reload this module and update
+    ``__class__`` of all suites it defines.
 
 * The interactive report is updated with any changed test cases.
 
+There are a few known limitations that reload doesn't always work under certain
+circumstances.
+
+* ``__main__`` module (i.e. test_plan.py) cannot be reloaded. Because of this,
+  it is recommended that test suites are defined in seperate modules. It is a
+  good idea to keep the main script simple.
+* Testplan is not able to reload module that is under *Namespace Package*, the
+  workaround is to add dummy __init__.py file.
+* Unlike testcases that can be added/removed/modified, adding/removing/renaming
+  testsuite class will not take effect after reloading.
+* Modules or objects imported using ``from some_module import *`` syntax are
+  not guaranteed to be properly reloaded. Such syntax is not encouraged in
+  Python anyway as it may bring naming conflicts.
+* In interactive mode, all modules containing test targets will be loaded at
+  the same time, so pay attention that avoid scheduling tasks having duplicate
+  module name but from different paths.
```

Comment:
If the modules from which tasks are scheduled have the same name, they won't reload properly.